test/ModAttachr/modAttachr.ts

```diff
@@ -0,0 +1,71 @@
+import { EventNames } from "../../src/EventNames";
+import { IMod } from "../../src/IModAttachr";
+import { ModAttachr } from "../../src/ModAttachr";
+import { mochaLoader } from "../main";
+import { CustomEventNames } from "../utils/fakes";
+
+mochaLoader.it("Testing to ensure onModEnable is fired properly", (): void => {
+    // Arrange
+    const eventNames: EventNames = new EventNames();
+    const dummyMod: IMod = {
+        name: "Dummy Mod",
+        events: {
+            onModEnable: (): string => {
```

Comment:
Hardcoding `onModEnable` as the key in the `events` object is equivalent to hardcoding `"onModEnable"` as a string literal. You should instead by using `[eventNames.onModEnable]: ...`.